test/tests/functional/pbs_cgroups_hook.py

```diff
@@ -1455,6 +1506,62 @@ def test_cgroup_cpuset(self):
                             'Processes should be assigned to different CPUs')
         self.logger.info('CpuIDs check passed')
 
+    def test_cgroup_cpuset_ncpus_are_cores(self):
+        """
+        Test to verify that correct number of jobs run on a hyperthread
+        enabled system when ncpus_are_cores is set to true.
+        """
+        # Check that system has hyperthreading enabled and has two processors
+        pcpus = 0
+        sibs = 0
+        cores = 0
+        pval = 0
+        phys = 1
+        prev = 0
+        with open('/proc/cpuinfo', 'r') as desc:
+            for line in desc:
+                if re.match('^processor', line):
+                    pcpus += 1
+                sibs_match = re.search(r'siblings	: ([1-9]+)', line)
+                cores_match = re.search(r'cpu cores	: ([1-9]+)', line)
+                phys_match = re.search(r'physical id	: ([1-9]+)', line)
+                if sibs_match:
+                    sibs = int(sibs_match.groups()[0])
+                if cores_match:
+                    cores = int(cores_match.groups()[0])
+                if phys_match:
+                    pval = int(phys_match.groups()[0])
+                    if pval != prev:
+                        prev = pval
+                        phys += 1
+        if (sibs == 0 or cores == 0):
+            self.skipTest('Insufficient information about the processors.')
+        if pcpus < 2:
+            self.skipTest('This test requires at least two processors.')
+        if sibs/cores == 1:
+            self.skipTest('This test requires hyperthreading to be enabled.')
+        name = 'CGROUP18'
+        self.load_config(self.cfg8 % ('', '', '', self.swapctl, ''))
+        # Submit M*N jobs, where M is the amount of physical processors and
+        # N is number of 'cpu cores' per M. Expect them to run.
+        njobs = phys * cores
+        a = {'Resource_List.select': '1:ncpus=1:mem=300mb:host=%s' %
+             self.hosts_list[0], ATTR_N: name + 'a'}
+        for j in range(njobs):
+            j = Job(TEST_USER, attrs=a)
```

Comment:
You are already using j as the iterator for the loop. If you don't use the value of the iterator, then replace it with an underscore as in...
```
for _ in range(njobs):
```